openqa-investigate

```diff
@@ -51,13 +51,24 @@ clone() {
     base_prio=$(echo "$clone_job_data" | runjq -r '.job.priority') || return $?
     clone_settings=("TEST+=:investigate$name_suffix" '_TRIGGER_JOB_DONE_HOOK=1' '_GROUP_ID=0' 'BUILD=')
     if [[ $refspec ]]; then
-	vars_json=$(fetch-vars-json "$origin")
+        vars_json=$(fetch-vars-json "$origin")
         testgiturl=$(echo "$vars_json" | runjq -r '.TEST_GIT_URL')
         casedir=$(echo "$clone_job_data" | runjq -r '.job.settings.CASEDIR') || return $?
         [[ $testgiturl != null ]] && casedir=$testgiturl
         [[ $casedir == null ]] && casedir=''
         repo=${casedir:-'https://github.com/os-autoinst/os-autoinst-distri-opensuse.git'}
         clone_settings+=("CASEDIR=${repo%#*}#${refspec}")
+        worker_class=$(echo "$vars_json" | runjq -r '.WORKER_CLASS') || return $?
+        used_worker=$(echo "$clone_job_data" | runjq -r '.job.settings.WORKER_CLASS') || return $?
+        mapfile -td ',' workers < <(echo "${worker_class}")
+        if [[ "${workers[@]}" =~ "$used_worker" ]]; then
+            if [[ $worker_class != null ]] && [[ $name_suffix =~ (last_good_test|last_good_build) ]]; then
```

Comment:
If we only need this logic for `last_good_test|last_good_build`, then this `if` should probably be the outer one

Reply:
There might have been a confusion. I also discussed with @b10n1k and I clarifed that *only* the combination job called "last_good_test+last_good_build" should be forced to run on the same worker as the original job. The jobs "last_good_test" *as well as* "last_good_build" should not be modified at all